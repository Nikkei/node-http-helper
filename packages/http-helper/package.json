{
    "name": "@nikkei/http-helper",
    "version": "2.4.1",
    "type": "module",
    "files": [
        "__dist/",
        "LICENSE",
        "package.json",
        "README.md"
    ],
    "typings": "./__dist/mod.d.ts",
    "exports": {
        ".": {
            "types": "./__dist/mod.d.ts",
            "default": "./__dist/mod.js"
        },
        "./auth": {
            "types": "./__dist/auth/mod.d.ts",
            "default": "./__dist/auth/mod.js"
        },
        "./cache": {
            "types": "./__dist/cache/mod.d.ts",
            "default": "./__dist/cache/mod.js"
        },
        "./csp": {
            "types": "./__dist/csp/mod.d.ts",
            "default": "./__dist/csp/mod.js"
        },
        "./sec_fetch": {
            "types": "./__dist/sec_fetch/mod.d.ts",
            "default": "./__dist/sec_fetch/mod.js"
        }
    },
    "scripts": {
        "clean": "npm run clean:dist && npm run clean:tsbuildinfo",
        "clean:dist": "rm -rf __dist/",
        "clean:tsbuildinfo": "rm -rf tsconfig.tsbuildinfo",
        "prebuild": "npm run clean",
        "build": "tsc --project ."
    },
    "repository": {
        "type": "git",
        "url": "git+https://github.com/Nikkei/node-http-helper.git"
    },
    "author": "Nikkei Inc.",
    "license": "MIT",
    "bugs": {
        "url": "https://github.com/Nikkei/node-http-helper/issues"
    },
    "homepage": "https://github.com/Nikkei/node-http-helper",
    "dependencies": {
        "option-t": "^39.0.2"
    }
}
