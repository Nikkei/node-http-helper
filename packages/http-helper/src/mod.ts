export * as Auth from './auth/mod.js';
export * as Cache from './cache/mod.js';
export * as ClearSiteData from './clear_site_data.js';
export * as CSP from './csp/mod.js';
export * as HttpHeaderName from './http_header_name.js';
export * as HttpMethod from './http_method.js';
export * as HttpStatus from './http_status_code.js';
export * as Mime from './mime.js';
export { SecFetchDest, SecFetchMode, SecFetchSite, SecFetchUser } from './sec_fetch/mod.js';
