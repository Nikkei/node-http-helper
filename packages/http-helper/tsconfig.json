{
    "extends": "./tsconfig.base.json",
    "compilerOptions": {
        "outDir": "./__obj"
    },
    "include": [
        // @prettier-ignore
        "src/**/*",
        "__tests__/**/*"
    ]
}
