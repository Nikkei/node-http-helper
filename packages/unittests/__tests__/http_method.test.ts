import { test, expect } from 'vitest';

import { HttpMethod } from '@nikkei/http-helper';

test('HttpMethod exported items', () => {
    expect(HttpMethod).toMatchSnapshot();
});
