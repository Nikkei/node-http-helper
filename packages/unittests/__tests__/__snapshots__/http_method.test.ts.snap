// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HttpMethod exported items 1`] = `
{
  "CONNECT": "CONNECT",
  "DELETE": "DELETE",
  "GET": "GET",
  "HEAD": "HEAD",
  "OPTIONS": "OPTIONS",
  "PATCH": "PATCH",
  "POST": "POST",
  "PUT": "PUT",
  "TRACE": "TRACE",
}
`;
